rust-runtime/aws-smithy-client/src/builder.rs

```diff
@@ -308,54 +332,92 @@ mod tests {
         }
     }
 
-    const TIMEOUTS_WITHOUT_SLEEP_MSG: &str =
-        "One or more timeouts were set, but no `sleep_impl` was passed into the builder";
-    const RETRIES_WITHOUT_SLEEP_MSG: &str =
-        "Retries require a `sleep_impl`, but none was passed into the builder.";
-    const RECOMMENDATION_MSG: &str =
-        "consider using the `aws-config` crate to load a shared config";
+    #[test]
+    fn defaults_dont_panic() {
+        let builder = Builder::new()
+            .connector(NeverConnector::new())
+            .middleware(tower::layer::util::Identity::new());
+
+        let _ = builder.build();
+    }
 
     #[test]
-    #[tracing_test::traced_test]
-    fn sleep_impl_given_no_warns() {
-        let _client = Builder::new()
+    fn defaults_panic_if_default_tokio_sleep_not_available() {
+        let mut builder = Builder::new()
             .connector(NeverConnector::new())
-            .middleware(tower::layer::util::Identity::new())
-            .sleep_impl(Some(Arc::new(StubSleep)))
-            .build();
+            .middleware(tower::layer::util::Identity::new());
+        builder.set_sleep_impl(None);
 
-        assert!(!logs_contain(TIMEOUTS_WITHOUT_SLEEP_MSG));
-        assert!(!logs_contain(RETRIES_WITHOUT_SLEEP_MSG));
-        assert!(!logs_contain(RECOMMENDATION_MSG));
+        let result = panic::catch_unwind(AssertUnwindSafe(move || {
+            let _ = builder.build();
+        }));
```

Comment:
Is this preferred to using `#[should_panic = "the panic message"]`?

Reply:
I wanted to ensure the panic originated from the `build` method, which the `catch_unwind` allows me to do.